ow_faults/src/FaultInjector.cpp

```diff
@@ -19,8 +19,10 @@ FaultInjector::FaultInjector(ros::NodeHandle node_handle)
   m_joint_state_pub = node_handle.advertise<sensor_msgs::JointState>("/joint_states", 10); 
 
   //power fault publishers and subs
-  m_fault_power_state_of_charge_pub = node_handle.advertise<std_msgs::Float64>("temporary/power_fault/state_of_charge", 10);
-  m_fault_power_temp_pub = node_handle.advertise<std_msgs::Float64>("temporary/power_fault/temp_increase", 10);
+  m_power_soc_sub = node_handle.subscribe("/_original/power_system_node/state_of_charge", 10, &FaultInjector::powerSOCListener, this);
+  m_fault_power_state_of_charge_pub = node_handle.advertise<std_msgs::Float64>("/power_system_node/state_of_charge", 10);
+  m_fault_power_temp_pub = node_handle.advertise<std_msgs::Float64>("/temporary/power_fault/temp_increase", 10);
+  timerPublishTemperature = node_handle.createTimer(ros::Duration(60), std::bind(&FaultInjector::powerFaultCb, this));
```

Comment:
I don't think you need a timer for our case here
```suggestion
```

Reply:
We can instead just use the original topic callback as the trigger to publish a new message

Reply:
I believe we had discussed in the dev day that they wanted the publish rate to be 60 seconds. I don't know what the original topic had as the rate and I believe that will be updated since Jason's PR removed publishing SOC altogether. We will need to wait until the new PR to see what the rate it. If their rate is also 1 minute, that would make your approach reasonable. If not, then we would still need to trigger a 60s publish rate ourselves.

Reply:
If the original topic currently doesn't  have the correct publish rate then it should be fixed by the originating node, not corrected by the faults injection node. When no faults injected your node should just relay the information as is with no delay.

Reply:
fair enough, will update it to be no timer. I'm also going to preset the soc to be a value because otherwise jason's PR would break this work.